src/libsystemd/sd-id128/id128-util.c

```diff
@@ -128,12 +128,18 @@ int id128_read_fd(int fd, Id128Format f, sd_id128_t *ret) {
         return sd_id128_from_string(buffer, ret);
 }
 
-int id128_read(const char *p, Id128Format f, sd_id128_t *ret) {
+int id128_read_prefix(const char *p, Id128Format f, sd_id128_t *ret, const char *root) {
```

Comment:
please keep return params last, i.e. swap the two last args, or maybe even put the root arg second